lib/mix/lib/mix/scm/git.ex

```diff
@@ -155,16 +156,25 @@ defmodule Mix.SCM.Git do
     if branch = opts[:branch] do
       "origin/#{branch}"
     else
-      opts[:ref] || opts[:tag] || "origin/master"
+      opts[:ref] || opts[:tag]
     end
   end
 
-  defp get_rev_info do
-    destructure [origin, rev],
-      :os.cmd('git --git-dir=.git config remote.origin.url && git --git-dir=.git rev-parse --verify --quiet HEAD')
+  defp get_rev_info(opts) do
+    cmd = 'git --git-dir=.git config remote.origin.url && git --git-dir=.git rev-parse --verify --quiet HEAD'
```

Comment:
Actually, it doesn't make sense. We always need to check HEAD because that's where we actually point to. There is something fishy about this still, let me see if I can put a finger on it.